src/runtime/threading_backend.cc

```diff
@@ -66,7 +138,16 @@ class ThreadGroup::Impl {
 #endif
 #if defined(__linux__) || defined(__ANDROID__)
     for (unsigned i = 0; i < threads_.size(); ++i) {
-      unsigned core_id = i + exclude_worker0;
+      unsigned core_id;
+      if (sorted_order_.size() >= threads_.size()) {
```

Comment:
again this seems to be not necessary. Can we simply do CHECK_EQ(sorted_order_.size(), threads_.size()); and always produce  sorted_ order before call SetAffinity? Possibly do sorted order initialization in the construction phase.

This will likely simplify this function's logic.